From 6506d607a0d6cfeb3e9e4de1d9f20cadafafb1d6 Mon Sep 17 00:00:00 2001
From: Rahul Pon <theflyingrahul@gmail.com>
Date: Sat, 26 Sep 2026 20:39:42 +0000
Subject: [PATCH] lscpu: add Qualcomm Oryon-2 part ID

Snapdragon X2 Elite cores report MIDR implementer 0x51 with part 0x002
(variants 0x1 and 0x2), which lscpu shows as "-". Linux's glymur.dtsi
describes these cores as "qcom,oryon-2-1" and "qcom,oryon-2-2",
following the "qcom,oryon-1-N" naming of the X1 Elite's part 0x001,
listed here as "Oryon".

A part ID can cover several SKUs, and the firmware names the exact one
in the SMBIOS processor version, for example "Snapdragon(R) X2 Elite -
X2E84100 - Qualcomm Oryon(TM) CPU". Prefer that name when it mentions
Snapdragon, as for Kunpeng.

Seen on an HP OmniBook 5 16-bf1xxx (X2E84100): CPUs 0-5 MIDR
0x512f0021, CPUs 6-11 MIDR 0x511f0021.

Signed-off-by: Rahul Pon <theflyingrahul@gmail.com>
---
 sys-utils/lscpu-arm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sys-utils/lscpu-arm.c b/sys-utils/lscpu-arm.c
index cda86f874..84b5d1665 100644
--- a/sys-utils/lscpu-arm.c
+++ b/sys-utils/lscpu-arm.c
@@ -157,6 +157,7 @@ static const struct id_part apm_part[] = {
 
 static const struct id_part qcom_part[] = {
     { 0x001, "Oryon" },
+    { 0x002, "Oryon-2", FL_FOLLOW_BIOS, "Snapdragon" },
     { 0x00f, "Scorpion" },
     { 0x02d, "Scorpion" },
     { 0x04d, "Krait" },
-- 
2.53.0

